samtranslator/model/sam_resources.py

```diff
@@ -2064,3 +2066,39 @@ def _add_connector_metadata(
             except KeyError:
                 original_metadata = {}
             resource.set_resource_attribute("Metadata", {**original_metadata, **metadata})
+
+
+class SamGraphQLApi(SamResourceMacro):
+    """SAM GraphQL API Macro (WIP)."""
+
+    resource_type = "AWS::Serverless::GraphQLApi"
+    property_types = {
+        "Name": Property(False, IS_STR),
+        "Tags": Property(False, IS_DICT),
+        "XrayEnabled": PassThroughProperty(False),
+        "Auth": Property(True, IS_DICT),
+    }
+
+    Auth: Auth
+    Tags: Optional[Dict[str, Any]]
+    XrayEnabled: Optional[PassThrough]
+    Name: Optional[str]
+
+    @cw_timer
+    def to_cloudformation(self, **kwargs: Any) -> List[Resource]:  # type: ignore
+        appsync_api = self._construct_appsync_api()
+        resources: List[Resource] = [appsync_api]
+
+        return resources
+
+    def _construct_appsync_api(self) -> GraphQLApi:
+        api = GraphQLApi(logical_id=self.logical_id, depends_on=self.depends_on, attributes=self.resource_attributes)
+
+        api.AuthenticationType = self.Auth["Type"]
```

Comment:
How does it fail when there's no `Type`? Want to make sure it's a customer error, and not a `KeyError`.

Reply:
Me and Slava talked about this [here](https://github.com/aws/serverless-application-model/pull/2803#discussion_r1084744398), we're going to make a more resilient solution to catch errors in the `Auth` object in a followup PR.